adal/src/main/java/com/microsoft/aad/adal/BasicWebViewClient.java

```diff
@@ -172,7 +172,7 @@ public void run() {
                                 new JWSBuilder());
                         final ChallengeResponse challengeResponse = certHandler
                                 .getChallengeResponseFromUri(challengeUrl);
-                        final HashMap<String, String> headers = new HashMap<String, String>();
+                        final HashMap<String, String> headers = new HashMap<>();
```

Comment:
I know it's not your change, but since we are already here, it's not a good practice in java to have the variable type as a concrete Collection type, should just be Map